### PR TITLE
Completed hw2

### PR DESCRIPTION
**Pull request by imOnVacation**

### CODE REVIEW

File: hw2/01-coins.js [hunk 1]
```diff
@@ -1,15 +1,77 @@
...
+                str = `${str + arr[3]} nickel, `
+            } else {
+                str = `${str + arr[3]} nickels, `
+            }
+        }
+        if (arr[4] !== 0) {
+            if (arr[4] === 1) {
+                str = `${str + arr[4]} penny`
+            } else {
+                str = `${str + arr[4]} pennies`
```
Comment (vuong2-pdx) [inline hw2/01-coins.js:55]: I like your use of template literals here

File: hw2/02-reverse.js [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+function Reverse() {
+    const input = document.getElementById('input').value
+    if (input.toString().length !== 8) {
+        document.getElementById('output').innerHTML = 'Error: Please input an 8-digit number'
+    } else {
+        const num = input.split('').reverse().join('')
+        const str = `${input} --> ${num}`
+        document.getElementById('output').innerHTML = str
+    }
+}
```
Comment (vuong2-pdx) [inline hw2/02-reverse.js:15]: If you wanted to match Caterina's screenshots, I think you need to somehow set the text color to red or green depending on the output

Comment (vuong2-pdx) [reply hw2/02-reverse.js:15]: I would approve your PR but I am not added to your repository. Anyway great job on this homework!

**Review by vuong2-pdx — APPROVED** [review]

File: hw2/02-reverse.js (hunk 2 above)
Comment (sashetov) [reply hw2/02-reverse.js:15]: Similar to what Tram is saying, you could set the color of the output element text to be different.
I personally did this by creating two different div elements, one with class "valid-feedback" and one with class "invalid-feedback", which will be green and red, respectively (are native bootstrap classes also) and would write to the appropriate div depending on the case

File: hw2/03-form.html [hunk 3]
```diff
@@ -14,6 +14,34 @@
   <body class="bg-dark">
     <form class="bg-light border rounded w-50 mx-auto mt-5 p-3">
       <h1 class="mt-2 mb-4">Contact Form</h1>
+      <div class="form-group">
+        <label for="inputName">Name *</label>
+        <input type="name" class="form-control" id="name" required />
+        <label for="inputEmail">Email *</label>
+        <input type="email" class="form-control" id="email" required />
```
Comment (sashetov) [inline hw2/03-form.html:21]: good use of input types

File: hw2/04-api.js [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
+    fetch(url)
+        .then((response) => {
+            return response.json()
+        })
+        .then((data) => {
+            for (let i = 0; i < data.length; i += 1) {
+                pop = numFormat.format(data[i].population)
+                map.set(data[i].name.common, pop)
+            }
+            const mapAsc = new Map([...map.entries()].sort())
```
Comment (sashetov) [inline hw2/04-api.js:25]: I think this should work but I'm not clear if it works in cases where the text is something like "Åland Islands". To cover that case I had to use the localeCompare function of the String class in javascript as the comparator function for sort.

**Review by sashetov — APPROVED** [review]
I did test it and apart from the issue with "Åland Islands" in sorting i think its good to merge. Locale based sorting was not stated as an explicit requirement so I wouldn't request that change but if you want to match the output exactly as Paun did it, you would need to implement something similar.

File: hw2/02-reverse.html [hunk 5]
```diff
@@ -23,8 +23,10 @@ <h1 class="mt-2 mb-4">Reverse</h1>
           class="btn btn-primary"
           value="Reverse"
           id="reverse"
+          onclick="Reverse()"
         />
       </div>
+      <p><span id="output"></span></p>
```
Comment (prachi-kashyap) [inline hw2/02-reverse.html:29]: Maybe you can add a break statement over here to get the same look and feel as Caterina's example screenshot with the text color changes.

File: hw2/04-api.js (hunk 4 above)
Comment (prachi-kashyap) [reply hw2/04-api.js:25]: I was stuck in this program and got an idea of how to print the data. Great work!

**Pull request merged**